kyuubi-events/src/main/scala/org/apache/kyuubi/events/handler/KafkaLoggingEventHandler.scala

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.events.handler
+
+import java.util.Properties
+
+import org.apache.kafka.clients.producer.{KafkaProducer, ProducerRecord}
+
+import org.apache.kyuubi.Logging
+import org.apache.kyuubi.config.KyuubiConf
+import org.apache.kyuubi.events.KyuubiEvent
+
+/**
+ * This event logger logs Kyuubi engine events to Kafka.
+ */
+class KafkaLoggingEventHandler(
+    topic: String,
+    producerConf: Properties,
+    kyuubiConf: KyuubiConf) extends EventHandler[KyuubiEvent]
+  with Logging {
+
+  private val kafkaProducer: KafkaProducer[String, String] = {
```

Comment:
when to close?

Reply:
`KafkaLoggingEventHandler` doesn't seem to implement the `AutoCloseable` interface. Did I miss something?

Reply:
Oh, thanks for the catch. That's my mistake as I added and removed the closing mechanism several times for repeating unstable test results in random ut cases and I missed it the last time. Made `KafkaLoggingEventHandler` implement `AutoCloseable` again now.

Reply:
Don't know why when I close the handler in `KafkaLoggingEventHandler`. It fails the other ut,
```
- get operation log *** FAILED ***
  scala.collection.JavaConverters.asScalaBufferConverter[String](logRowSet.getLogRowSet()).asScala.exists(((x$1: String) => x$1.contains("show tables"))) was false (OperationsResourceSuite.scala:103)
```

Reply:
Resolve this conversation as https://github.com/apache/kyuubi/pull/4777 close event handlers when stopping server.